ui/pages/confirmations/confirm-approve/confirm-approve.util.js

```diff
@@ -21,16 +21,14 @@ export function getCustomTxParamsData(
   if (spender.startsWith('0x')) {
     spender = spender.substring(2);
   }
-  const [signature, tokenValue] = data.split(spender);
-
-  if (!signature || !tokenValue) {
+  const [signature, rest] = data.split(spender);
+  if (!signature || !rest) {
     throw new Error('Invalid data');
-  } else if (tokenValue.length !== 64) {
```

Comment:
Perhaps we ought to continue to throw if `rest` value is less than 64 characters?

Reply:
Could we perhaps add the logic here just in case that upstream code changes?

Reply:
Sure, just added this check back!